### PR TITLE
Refactoring: serialize message

### PR DESCRIPTION
**Pull request by ybensacq**
1. refactoring serialize function
2. add missing generic checksum
3. refactoring deserializeSlice function

### CODE REVIEW

**Review by tdelabro — COMMENTED** [review]
`deserializeSlice` can also be made generic

**Comment by tdelabro** [comment]
or removed, if it's never used .... But I think it's used for tests

File: src/network/protocol/messages/lib.zig [hunk 1]
```diff
@@ -168,3 +168,22 @@ pub fn genericChecksum(m: anytype) [4]u8 {
 
     return digest[0..4].*;
 }
+
+pub fn genericSerialize(m: anytype, allocator: std.mem.Allocator) ![]u8 {
```
Comment (tdelabro) [inline src/network/protocol/messages/lib.zig:172]: I thinmyou need comptime checks for `serializeToWriter` and `hintSerializedLen`

File: src/network/protocol/messages/lib.zig [hunk 2]
```diff
@@ -168,3 +168,22 @@ pub fn genericChecksum(m: anytype) [4]u8 {
...
+    const buffer = try allocator.alloc(u8, serialized_len);
+    errdefer allocator.free(buffer);
+
+    var fbs = std.io.fixedBufferStream(buffer);
+    try m.serializeToWriter(fbs.writer());
+
+    return buffer;
+}
+
+pub fn genericDeserializeSlice(comptime T: type, allocator: std.mem.Allocator, bytes: []const u8) !T {
```
Comment (tdelabro) [inline src/network/protocol/messages/lib.zig:184]: same here for `deserializeReader`

**Review by tdelabro — CHANGES_REQUESTED** [review]

File: src/network/protocol/messages/lib.zig (hunk 1 above)
Comment (ybensacq) [reply src/network/protocol/messages/lib.zig:172]: yes right!

**Review by tdelabro — APPROVED** [review]

**Pull request merged**